src/coreclr/jit/gentree.cpp

```diff
@@ -6791,18 +6788,25 @@ ExceptionSetFlags GenTree::OperExceptions(Compiler* comp)
         case GT_HWINTRINSIC:
         {
             GenTreeHWIntrinsic* hwIntrinsicNode = this->AsHWIntrinsic();
-            assert(hwIntrinsicNode != nullptr);
-            if (hwIntrinsicNode->OperIsMemoryLoadOrStore())
+
+            if (hwIntrinsicNode->OperIsMemoryLoad())
+            {
+                assert((gtFlags & GTF_IND_NONFAULTING) == 0);
+
+                // TODO-CQ: We should use comp->fgAddrCouldBeNull on the address operand
+                // to determine if this can actually produce an NRE or not
+
+                return ExceptionSetFlags::NullReferenceException;
+            }
+            else if (hwIntrinsicNode->OperIsMemoryStore())
             {
-                // This operation contains an implicit indirection
-                //   it could throw a null reference exception.
-                //
```

Comment:
I am not sure I understand the point of this change. Was it to add the `TODO-CQ`? If so, it would apply to stores as well as loads.

Reply:
Ah right, the reason we don't have GT_STOREIND above is because that's a lowering only thing...

Will fix to have the todo under the shared path.

Reply:
Fixed.